src/box/memtx_engine.h

```diff
@@ -188,6 +188,14 @@ struct memtx_engine {
 	int sort_threads;
 };
 
+/** Data for initialization of the join context. */
+struct memtx_prepare_data {
+	/** Vclock to respond with. */
+	struct vclock *vclock;
+	/** Whether sending JOIN_META stage is required. */
+	bool send_meta;
+};
```

Comment:
But actually the only "engine-independent" part is `cursor`. `send_meta` and `vclock` are related only to memtx engine, vinyl doesn't need them at all.

Reply:
By "engine-independent" I mean that it can be set pr used outside the engine. Both `send_meta` and `vclock` can. It's okay that they aren't used by Vinyl.

Reply:
Indeed, misunderstood you, it's better this way. Moreover, we don't need to allocate contexts of the engines for checkpoint join. Fixed.